app/main.py

```diff
@@ -1 +1,31 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (
+            "{"
+            + f"Name: {self.name}, "
+              f"Health: {self.health}, "
+              f"Hidden: {self.hidden}"
+            + "}"
+        )
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(herb: Herbivore) -> None:
+        if type(herb) is Herbivore and not herb.hidden:
```

Comment:
I think it's better to use isinstance() method here for this purpose

Reply:
> I think it's better to use isinstance() method here for this purpose

what if we inherit class from carnivore?